wikispeedruns/leaderboards.py

```diff
@@ -25,6 +25,51 @@ def get_db():
     return conn
 '''
 
+LEADERBOARD_COLUMNS = ['run_id',
+        'start_time',
+        'end_time',
+        'play_time',
+        'finished',
+        'path',
+        'prompt_id',
+        'user_id',
+        'username',
+        'rank',
+        'numRuns',
+        'path_length']
+
+BASE_ALIAS = 'runs'
+OUTER_ALIAS = 'processed_runs'
+
+def _query_select_columns(table):
+    separator = ', '
+    if __debug__:
+        separator = ',\n\t'
+
+    return separator.join([f'{table}.{column}' for column in LEADERBOARD_COLUMNS])
+
+def _query_current_run_clause(table, run_id=None):
+    if run_id is None:
+        return '0'
+
+    return f'({table}.run_id = %(run_id)s AND {table}.prompt_id = %(prompt_id)s)'
```

Comment:
The only practical change in this diff comes from this line, where the change is
`current_run_clause = 'runs.run_id = %(run_id)s'` 
to
`current_run_clause = '(runs.run_id = %(run_id)s' AND runs.prompt_id = %(prompt_id)s)`